Framework/DataHandling/inc/MantidDataHandling/LoadBankFromDiskTask.h

```diff
@@ -54,12 +54,13 @@ class MANTID_DATAHANDLING_DLL LoadBankFromDiskTask : public Kernel::Task {
 
 private:
   void loadPulseTimes(::NeXus::File &file);
-  void loadEventIndex(::NeXus::File &file, std::vector<uint64_t> &event_index);
+  std::unique_ptr<std::vector<uint64_t>> loadEventIndex(::NeXus::File &file);
   void prepareEventId(::NeXus::File &file, int64_t &start_event,
-                      int64_t &stop_event, std::vector<uint64_t> &event_index);
-  void loadEventId(::NeXus::File &file);
-  void loadTof(::NeXus::File &file);
-  void loadEventWeights(::NeXus::File &file);
+                      int64_t &stop_event,
+                      const std::vector<uint64_t> &event_index);
+  std::unique_ptr<uint32_t[]> loadEventId(::NeXus::File &file);
+  std::unique_ptr<float[]> loadTof(::NeXus::File &file);
+  std::unique_ptr<float[]> loadEventWeights(::NeXus::File &file);
```

Comment:
similarly, can these three functions return a `std::vector`? Or does that expand the scope too much?

Reply:
I had considered this at first too but then later on the array is shared between several parallel tasks and refactoring that code was going to expand the scope a little too much.